src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -1986,8 +1987,9 @@ void Creature::setDeathState(DeathState s)
         if (m_formation && m_formation->GetLeader() == this)
             m_formation->FormationReset(true);
 
-        bool needsFalling = IsFlying() || IsHovering();
-        SetHover(false);
+        bool needsFalling = (IsFlying() || IsHovering()) && !IsUnderWater();
+        SetHover(false, false);
+        SetDisableGravity(false, false);
```

Comment:
Why

Reply:
This one is present in your 434 commit as well. https://github.com/TrinityCore/TrinityCore/commit/5cb747620a7cb897135650c519ca2daa50523f40#diff-f3f953a63e2a008d22de69cb68f735c4R1827
:shrug:

Reply:
Well that's an oops? @Ovahlord unless this was the corpse falling issue? I forgot already

Reply:
pretty sure it was because of corpse falling, ye